common/fixation_entry.go

```diff
@@ -83,12 +129,36 @@ func entryKey(
 	return key
 }
 
+func deleteStaleEntries(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string, cdc codec.BinaryCodec, index string) {
+	// get current block
+	currentBlock := ctx.BlockHeight()
+
+	// get all the entries of the entry's index
+	entryList := GetAllEntriesForIndex(ctx, storeKey, entryKeyPrefix, cdc, index)
+
+	// iterate over the entries
+	for _, entry := range entryList {
+		// if the entry's references is zero and the entry's block + STALE_ENTRY_TIME (the time it takes for an entry to be stale) is smaller than the current block, remove it
+		if entry.GetReferences() == 0 && int64(entry.GetBlock())+types.STALE_ENTRY_TIME < currentBlock {
+			RemoveEntry(ctx, storeKey, cdc, entryKeyPrefix, index)
+		}
+	}
+}
+
 // Function to create a new fixation entry, add it to the KVStore and update the entry's older versions indices. Note, the entryIndex should be without the version num suffix
 func AddEntry(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string, uniqueIndexEntryKeyPrefix string, cdc codec.BinaryCodec, entryIndex string, marshalledData []byte) error {
+	// delete stale entries
+	deleteStaleEntries(ctx, storeKey, entryKeyPrefix, cdc, entryIndex)
+
+	// since we're keeping versions with a "_<num>" suffix, to avoid trouble we forbid that the last character would be "_"
+	if entryIndex[len(entryIndex)-1:] == "_" {
```

Comment:
if we dont rely on strings we dont need this check

Reply:
done